File: articles/brain-development-plasticity-and-the-developing-nervous-system/cpp/brain_development_summary.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>
std::vector<std::string> split(const std::string& s){ std::vector<std::string> v; std::stringstream ss(s); std::string x; while(std::getline(ss,x,',')) v.push_back(x); return v; }
int idx(const std::vector<std::string>& h, const std::string& n){ for(size_t i=0;i<h.size();++i) if(h[i]==n) return (int)i; return -1; }
int main(int argc,char* argv[]){
  if(argc<2){ std::cerr<<"usage: brain_development_summary data/brain_development_panel.csv\n"; return 1; }
  std::ifstream file(argv[1]); std::string line; std::getline(file,line); auto h=split(line);
  std::vector<int> cols={idx(h,"neural_state"),idx(h,"developmental_outcome"),idx(h,"acute_stress"),idx(h,"developmental_support_context")};
  std::vector<std::string> names={"neural_state","developmental_outcome","acute_stress","developmental_support_context"};
  std::vector<double> sums(cols.size(),0.0); long n=0;
  while(std::getline(file,line)){ auto v=split(line); for(size_t j=0;j<cols.size();++j) sums[j]+=std::stod(v[cols[j]]); n++; }
  std::cout<<"Rows analyzed: "<<n<<"\n"; for(size_t j=0;j<names.size();++j) std::cout<<"Mean "<<names[j]<<": "<<sums[j]/n<<"\n"; return 0;
}

File: articles/brain-development-plasticity-and-the-developing-nervous-system/go/brain_development_summary.go
package main
import ("encoding/csv"; "fmt"; "log"; "os"; "strconv")
func idx(h []string, name string) int { for i, v := range h { if v == name { return i } }; return -1 }
func main() {
 if len(os.Args) < 2 { log.Fatal("usage: go run brain_development_summary.go data/brain_development_panel.csv") }
 f, err := os.Open(os.Args[1]); if err != nil { log.Fatal(err) }; defer f.Close()
 rows, err := csv.NewReader(f).ReadAll(); if err != nil { log.Fatal(err) }
 h := rows[0]
 cols := []int{idx(h,"neural_state"),idx(h,"developmental_outcome"),idx(h,"acute_stress"),idx(h,"developmental_support_context")}
 names := []string{"neural_state","developmental_outcome","acute_stress","developmental_support_context"}
 sums := make([]float64, len(cols))
 for _, r := range rows[1:] { for j, c := range cols { v, _ := strconv.ParseFloat(r[c], 64); sums[j] += v } }
 fmt.Printf("Rows analyzed: %d\n", len(rows)-1)
 for j, name := range names { fmt.Printf("Mean %s: %.4f\n", name, sums[j]/float64(len(rows)-1)) }
}
